### DISCUSSION TITLE
Exception:  [!] No espeak backend found. Install espeak-ng or espeak to your system.

### DISCUSSION THREAD

**Discussion by pnowakowski614** [coqui-ai/TTS#1973] [General Q&A]
Hi, I'm trying to synthesize speech on Windows 10 using blizzard2013/capacitron-t2-c50 tts model and blizzard2013/hifigan as a vocoder. I've installed espeak-ng (https://github.com/espeak-ng/espeak-ng/blob/master/docs/guide.md) but I still keep getting traceback as follows:
 
```Traceback (most recent call last):
  File "c:\users\piotrek\anaconda3\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\piotrek\anaconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\Piotrek\anaconda3\Scripts\tts.exe\__main__.py", line 7, in <module>
  File "c:\users\piotrek\anaconda3\lib\site-packages\TTS\bin\synthesize.py", line 309, in main
    synthesizer = Synthesizer(
  File "c:\users\piotrek\anaconda3\lib\site-packages\TTS\utils\synthesizer.py", line 76, in __init__
    self._load_tts(tts_checkpoint, tts_config_path, use_cuda)
  File "c:\users\piotrek\anaconda3\lib\site-packages\TTS\utils\synthesizer.py", line 113, in _load_tts
    self.tts_model = setup_tts_model(config=self.tts_config)
  File "c:\users\piotrek\anaconda3\lib\site-packages\TTS\tts\models\__init__.py", line 13, in setup_model
    model = MyModel.init_from_config(config, samples)
  File "c:\users\piotrek\anaconda3\lib\site-packages\TTS\tts\models\tacotron2.py", line 432, in init_from_config
    tokenizer, new_config = TTSTokenizer.init_from_config(config)
  File "c:\users\piotrek\anaconda3\lib\site-packages\TTS\tts\utils\text\tokenizer.py", line 188, in init_from_config
    phonemizer = get_phonemizer_by_name(config.phonemizer, **phonemizer_kwargs)
  File "c:\users\piotrek\anaconda3\lib\site-packages\TTS\tts\utils\text\phonemizers\__init__.py", line 42, in get_phonemizer_by_name
    return ESpeak(**kwargs)
  File "c:\users\piotrek\anaconda3\lib\site-packages\TTS\tts\utils\text\phonemizers\espeak_wrapper.py", line 91, in __init__
    raise Exception(" [!] No espeak backend found. Install espeak-ng or espeak to your system.")
Exception:  [!] No espeak backend found. Install espeak-ng or espeak to your system.
```

What may be the cause of this problem?

**Comment by p0p4k** [comment]
What does 'espeak-ng hello' in command prompt return?

**Reply by pnowakowski614** [reply]
It seems to be working - it returns an audio sample of the word.

**Reply by p0p4k** [reply]
If you see the [code](https://github.com/coqui-ai/TTS/blob/b95cf3363cca4563400cf95dd9fef75cf02a17b9/TTS/tts/utils/text/phonemizers/espeak_wrapper.py#L10-L22) it tries to find `espeak `or `espeak-ng` in your system using python's `which` util. You can probably try the following code in your current TTS environment to confirm if espeak is installed correctly, if not, you can try a fresh installation:
```
from shutil import which
print(which('espeak'))
print(which('espeak-ng'))
```
![image](https://user-images.githubusercontent.com/8834712/190275570-d0c1910e-09ee-4860-9354-62f90426fe3d.png)

**Reply by platinumshore** [reply]
might also need pip install python-espeak-ng aswell https://pypi.org/project/python-espeak-ng/

**Reply by SP1901** [reply]
Hi, I'm also having this issue, but I'm trying to use the voice tts_models/en/sam/tacotron-DDC, which also uses an espeak wrapper. I have used `pip install espeakng` and `pip install py-espeak-ng` in the anaconda3\envs\coquitts directory, but I cannot run `pip install python-espeak-ng` due to a legacy install failure (along with the message "This is an issue with the package mentioned above, not pip."). I am on Windows, using the coquitts environment. When I try to run `tts-server --model_name "tts_models/en/sam/tacotron-DDC"` I get an output very similar to the screenshot in the original question. I'd appreciate any suggestions for getting this working!

**Reply by Anomyous1** [reply]
```
pip install python-espeak-ng
Collecting python-espeak-ng
  Using cached python-espeak-ng-0.6.6.tar.gz (9.1 kB)
Using legacy 'setup.py install' for python-espeak-ng, since package 'wheel' is not installed.
Installing collected packages: python-espeak-ng
    Running setup.py install for python-espeak-ng ... error
    ERROR: Command errored out with exit status 1:

    .............

    espeakmodulecore.cpp
    espeak/espeakmodulecore.cpp(24): fatal error C1083: Cannot open include file: 'espeak-ng/speak_lib.h': No such file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\\tts\venv_tts\scripts\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\User\\AppData\\Local\\Temp\\pip-install-bo598j__\\python-espeak-ng_ae4c6eb2567f4e35b2164b72b4cc309d\\setup.py'"'"'; __file__='"'"'C:\\Users\\User\\AppData\\Local\\Temp\\pip-install-bo598j__\\python-espeak-ng_ae4c6eb2567f4e35b2164b72b4cc309d\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\User\AppData\Local\Temp\pip-record-d_dreko0\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\\tts\venv_tts\include\site\python3.9\python-espeak-ng' Check the logs for full command output.
```

**Reply by SophieDC98** [reply]
I ran !sudo apt-get install espeak in my Colab, restarted the runtime and everything is working fine now

**Reply by Corey4005** [reply]
> If you see the [code](https://github.com/coqui-ai/TTS/blob/b95cf3363cca4563400cf95dd9fef75cf02a17b9/TTS/tts/utils/text/phonemizers/espeak_wrapper.py#L10-L22) it tries to find `espeak `or `espeak-ng` in your system using python's `which` util. You can probably try the following code in your current TTS environment to confirm if espeak is installed correctly, if not, you can try a fresh installation:
> 
> ```
> from shutil import which
> print(which('espeak'))
> print(which('espeak-ng'))
> ```
> 
> ![image](https://user-images.githubusercontent.com/8834712/190275570-d0c1910e-09ee-4860-9354-62f90426fe3d.png)

This was actually my problem. I needed to download espeak using the [msi for windows](https://github.com/espeak-ng/espeak-ng/releases). Then, I added espeak-ng to my PATH variable and the code worked!

**Reply by fahnub** [reply]
@sophieDataroots solution worked for me.

**Reply by pepetikesavasiddhardha** [reply]
Yeah this is helping me thanks

**Comment by erdemkaymaz** [comment]
you can try `brew install espeak`, worked for me.

**Comment by FieldMarshallVague** [comment]
I get this same issue on windows.  I installed the espeak windows version using MSI installed and added install folder to path, but it looks like it doesn't see it, or more likely isn't looking for that install (the 'espeak-ng/speak_lib.h' header file isn't in that folder anyway, looks like it comes from a build package).

**Reply by FieldMarshallVague** [reply]
Ah, I misunderstood.  You actually have to run the EXE in the background, it's not being called from TTS.

So, you need to run the espeak.exe in background to run the ljs models.

e.g. Start-Process -FilePath "C:\Program Files\eSpeak NG\espeak-ng.exe" (in powershell for windows)

**Comment by liuyiyuan** [comment]
I met the same question in wins 11 sys;
I use cmd fix it;
"pip install python-espeak-ng"

**Comment by MissingTwins** [comment]
For colab
```
!apt update && apt install espeak 
!pip install espeakng 
# restart session
from shutil import which
print(which('espeak'))
print(which('espeak-ng'))
```

**Comment by Fatiprogramist** [comment]
same problem. what is solution for windows?

**Comment by stearno** [comment]
the pip install did not solve for me.

What fixed for me was put this at top of my code:
import os

os.environ["PATH"] = r"C:\Program Files\eSpeak NG;" + os.environ["PATH"]
